### PR TITLE
Confirmation links unit tests

### PR DESCRIPTION
**Pull request by AndriiHorishniiMOC**
**Jira issue:**
https://glia.atlassian.net/browse/MOB-2872

**What was solved?**
Added unit tests for confirmation dialog.

Also, the SnackBar tests now use SnapshotStringProvider. With these changes, the actual value of the R.string doesn't matter for the snapshots tests.

**Release notes:**

 - [ ] Feature
 - [x] Ignore
 - [ ] Release notes (Is it clear from the description here?)
 - [ ] Migration guide (If changes are needed for integrator already using the SDK - what needs to be communicated? Add underneath please)

**Additional info:**

 - [x] Tests updated? Added? Unit, acceptance, snapshots?
 - [ ] Logging for future troubleshooting of client issues added?

**Screenshots:**

### CODE REVIEW

File: widgetssdk/src/main/java/com/glia/widgets/callvisualizer/ActivityWatcherForCallVisualizerController.kt [hunk 1]
```diff
@@ -138,19 +142,13 @@ internal class ActivityWatcherForCallVisualizerController(
...
     }
 
-    override fun onLink1Clicked() {
-        Logger.d(TAG, "onLink1Clicked")
-        watcher.openWebBrowserActivity(
-            stringProvider.getRemoteString(R.string.engagement_confirm_link1_text),
-            stringProvider.getRemoteString(R.string.engagement_confirm_link1_url)
-        )
-    }
+    override fun getConfirmationDialogLinks() = confirmationDialogLinksUseCase(stringProvider)
```
Comment (Pelkar) [inline widgetssdk/src/main/java/com/glia/widgets/callvisualizer/ActivityWatcherForCallVisualizerController.kt:145]: I think that the usecase could already have the provider from creation

**Review by Pelkar — APPROVED** [review]
LGTM, left one comment

File: widgetssdk/src/main/java/com/glia/widgets/callvisualizer/ActivityWatcherForCallVisualizerController.kt (hunk 1 above)
Comment (AndriiHorishniiMOC) [reply widgetssdk/src/main/java/com/glia/widgets/callvisualizer/ActivityWatcherForCallVisualizerController.kt]: Fixed 187f4b051dc8862adfc04b296a92b3d75c7891be

**Review by DavDo — APPROVED** [review]
Overall LGTM.
Have you planned to add a snapshot test for the confirmation dialog in a separate PR?

**Comment by AndriiHorishniiMOC** [comment]
@DavDo Yes, I'm currently working on snapshot tests for the confirmation dialog.

**Comment by AndriiHorishniiMOC** [comment]
!squash

**Pull request merged**